File: NaoBlocks.Core/Models/Robot.cs
﻿using System;

namespace NaoBlocks.Core.Models
{
    public class Robot
    {
        public string FriendlyName { get; set; } = string.Empty;

        public string Id { get; set; } = string.Empty;

        public bool IsInitialised { get; set; }

        public string MachineName { get; set; } = string.Empty;

        public Password Password { get; set; } = Password.Empty;

        public string RobotTypeId { get; set; } = string.Empty;

        public RobotType? Type { get; set; }

        public DateTime WhenAdded { get; set; }
    }
}

File: Clients/Angular/src/app/data/robot.ts
import { NamedValue } from './named-value';
import { ParseResult } from './parse-result';
import { RobotLog } from './robot-log';
import { RobotType } from './robot-type';

export class Robot {
    id?: string;
    friendlyName?: string;
    machineName?: string;
    type?: string;
    password?: string;
    whenAdded?: Date;
    isNew: boolean;
    isLoading: boolean = false;
    isInitialised: boolean = false;
    logs: RobotLog[] = [];
    filteredLogs: RobotLog[] = [];

    typeDetails?: RobotType;
    parse?: ParseResult;
    values?: NamedValue[];

    constructor(isNew: boolean = false) {
        this.isNew = isNew;
    }
}
